ait/commons/util/command/upload.py

```diff
@@ -54,73 +105,37 @@ def get_files(upload_path, curr_path, level):
-                    f_name = os.path.basename(p)
-                    fs.append(FileTransfer(path=p, key=f_name, size=f_size))
+                    files.append(p)
 
                 elif os.path.isdir(p):  # recursively handle dir upload
                     get_files(p, p, 0)
 
-            def upload(idx):
-                try:
-                    key = selected_area + fs[idx].key
-
-                    # creating a new session for each file upload/thread, as it's unclear whether they're
-                    # thread-safe or not
-
-                    sess = self.aws.new_session()
-
-                    if not self.args.o and self.aws.obj_exists(key):
-                        fs[idx].status = 'File exists. Use -o to overwrite.'
-                        fs[idx].successful = True
-                        fs[idx].complete = True
-                    else:
-                        res = sess.resource('s3')
-                        ftype = filetype.guess(fs[idx].path)
-                        # default contentType
-                        contentType = 'application/octet-stream'
-                        if ftype is not None:
-                            contentType = ftype.mime
-                        contentType += '; dcp-type=data'
-                        
-                        # upload_file automatically handles multipart uploads via the S3 Transfer Manager
-                        # put_object maps to the low-level S3 API request, it does not handle multipart uploads
-                        res.Bucket(self.aws.bucket_name).upload_file(Filename=fs[idx].path, Key=key,
-                                                                     Callback=TransferProgress(fs[idx]),
-                                                                     ExtraArgs={'ContentType': contentType})
-
-                        # if file size is 0, callback will likely never be called
-                        # and complete will not change to True
-                        # hack
-                        if fs[idx].size == 0:
-                            fs[idx].status = 'Empty file.'
-                            fs[idx].complete = True
-                            fs[idx].successful = True
-
-                except Exception as thread_ex:
-                    fs[idx].status = 'Upload failed.'
-                    fs[idx].complete = True
-                    fs[idx].successful = False
-
             print('Uploading...')
-            transfer(upload, fs)
-
-            self.files = [f for f in fs if f.successful]
 
-            if all([f.successful for f in fs]):
-                return True, 'Successful upload.'
-
-            else:
-                return False, 'Failed upload.'
+            success = self.upload_files(files, selected_area)
+            return (success, "Successful upload") if success else (success, "Failed upload")
 
         except Exception as e:
             return False, format_err(e, 'upload')
+
+
+class ProgressBar:
```

Comment:
If this is going to be used elsewhere, then put it in another file?

Reply:
that's a good idea. I was thinking of leaving it this way till I refactored the rest of the project to use it. Currently the remaining code files use the classes defined in `file_transfer.py` for tracking progress, but it would be good to have `ProgressBar` defined in a separate file from the start